crates/bevy_scene/src/scene_loader.rs

```diff
@@ -35,7 +35,9 @@ impl AssetLoader for SceneLoader {
             let scene_deserializer = SceneDeserializer {
                 type_registry: &self.type_registry.read(),
             };
-            let scene = scene_deserializer.deserialize(&mut deserializer)?;
+            let scene = scene_deserializer
+                .deserialize(&mut deserializer)
+                .map_err(|e| deserializer.span_error(e))?;
```

Comment:
I ran into this myself and came up with a similar solution but I had it logging the file name as well:

```suggestion
            let scene = scene_deserializer
                .deserialize(&mut deserializer)
                .map_err(|e| {
                    let span_error = deserializer.span_error(e);
                    anyhow!(
                        "{} at {}:{}",
                        span_error.code
                        load_context.path().to_string_lossy(),
                        span_error.position,
                    )
                })?;
```

Have to import the anyhow macro e.g. `use anyhow::anyhow;` (Note that we are already converting to anyhow's Error anyway)

Should give an error message like: 

```
WARN bevy_asset::asset_server: encountered an error while loading an asset: Expected struct `Vec3` but found `a` at scenes/test/scene.scn.ron:10:4
```

Just a suggestion! What you have it already a huge improvement so please don't hold it up over this if what I have is at all controversial.

Reply:
yeah, this looks better. I was worried we needed to add anyhow to the Cargo.toml, but it's already used in bevy_scene.